src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -318,6 +321,209 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
+        } else
+        {
+            return node.CreateBindingFailure(HttpDiagnostics.CannotResolveSymbol(variableName));
+        }
+
+        
     }
+
+    private HttpBindingResult<object?>? MatchExpressionValue(HttpExpressionNode node, string expression)
+    {
+
+        var guidPattern = new Regex(@$"\{"$guid"}", RegexOptions.Compiled);
+        var dateTimePattern = new Regex(@$"\{"$datetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var localDateTimePattern = new Regex(@$"\{"$localDatetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var randomIntPattern = new Regex(@$"\{"$randomInt"}(?:\s(?<parameters>-?\d+)){{0,2}}", RegexOptions.Compiled);
+        var timestampPattern = new Regex(@$"\{"$timestamp"}(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+
+        var guidMatches = guidPattern.Matches(expression);
+        var dateTimeMatches = dateTimePattern.Matches(expression);
+        var localDateTimeMatches = localDateTimePattern.Matches(expression);
+        var randomIntMatches = randomIntPattern.Matches(expression);
+        var timestampMatches = timestampPattern.Matches(expression);
+
+        if (guidMatches.Count > 0)
+        {
+            return node.CreateBindingSuccess(Guid.NewGuid().ToString());
+        } else if(dateTimeMatches.Count > 0)
+        {
+            return GetDateTime(node, expression, dateTimeMatches, dateTimePattern);
+        } else if(localDateTimeMatches.Count > 0)
+        {
+            return GetDateTime(node, expression, localDateTimeMatches, localDateTimePattern);
+        } else if(randomIntMatches.Count > 0)
+        {
+            return GetRandInt(node, expression, randomIntMatches);
+        }else if(timestampMatches.Count > 0)
+        {
+            return GetTimestamp(node, expression, timestampMatches);
+        }
+        return null;
+    }
+
+    private HttpBindingResult<object?> GetTimestamp(HttpExpressionNode node, string expression, MatchCollection matches)
+    {
+        var text = expression;
+        var currentDateTimeOffset = DateTimeOffset.UtcNow;
+        
+        DateTimeOffset dateTimeOffset = currentDateTimeOffset;
+        var match = matches.FirstOrDefault();
+        if (match?.Groups.Count == 3)
+        {
+            try
+            {
+                if (match.Groups["offset"].Success && match.Groups["option"].Success
+                    && int.TryParse(match.Groups["offset"].Value, out int offset)
+                    && offset != 0)
+                {
+                    dateTimeOffset = dateTimeOffset.AddOffset(offset, match.Groups["option"].Value);
+                }
+
+                text = string.Concat(text.Substring(0, match.Index), dateTimeOffset.ToUnixTimeSeconds().ToString(), text.Substring(match.Index + match.Value.Length));
```

Comment:
I am not sure I understand why the `string.Concat` call is needed here. Shouldn't text just be set to the `dateTimeOffset.ToUnixTimeSeconds().ToString()` (so that `node.CreateBindingSuccess()` returns the corresponding value)?

Is this code that we forgot to change after the for loop from the old implementation was removed?